initrd/bin/tpmr

```diff
@@ -650,15 +672,15 @@ tpm2_reset() {
 	# output TPM Owner Password to a file to be reused in this boot session until recovery shell/reboot
 	DEBUG "Caching TPM Owner Password to $SECRET_DIR/tpm_owner_password"
 	echo -n "$tpm_owner_password" >"$SECRET_DIR/tpm_owner_password"
-	tpm2 clear -c platform || warn "Unable to clear TPM on platform hierarchy"
-	tpm2 changeauth -c owner "$(tpm2_password_hex "$tpm_owner_password")"
-	tpm2 changeauth -c endorsement "$(tpm2_password_hex "$tpm_owner_password")"
+	tpm2 clear -c platform >/dev/null 2>&1 || LOG "Unable to clear TPM on platform hierarchy"
+	tpm2 changeauth -c owner "$(tpm2_password_hex "$tpm_owner_password")" >/dev/null 2>&1 || LOG "Unable to change owner password"
+	tpm2 changeauth -c endorsement "$(tpm2_password_hex "$tpm_owner_password")" >/dev/null 2>&1 || LOG "Unable to change endorsement password"
 	tpm2 createprimary -C owner -g sha256 -G "${CONFIG_PRIMARY_KEY_TYPE:-rsa}" \
-		-c "$SECRET_DIR/primary.ctx" -P "$(tpm2_password_hex "$tpm_owner_password")"
+		-c "$SECRET_DIR/primary.ctx" -P "$(tpm2_password_hex "$tpm_owner_password")" >/dev/null 2>&1 || LOG "Unable to create primary key"
 	tpm2 evictcontrol -C owner -c "$SECRET_DIR/primary.ctx" "$PRIMARY_HANDLE" \
-		-P "$(tpm2_password_hex "$tpm_owner_password")"
-	shred -u "$SECRET_DIR/primary.ctx"
-	tpm2_startsession
+		-P "$(tpm2_password_hex "$tpm_owner_password")" >/dev/null 2>&1 || LOG "Unable to evict primary key"
+	shred -u "$SECRET_DIR/primary.ctx" >/dev/null 2>&1
+	tpm2_startsession >/dev/null 2>&1 || LOG "Unable to start session"
```

Comment:
TPM2 reset is not working on L1UM v2 :thinking:   Tried to do OEM reset and just get an error about TPM reset failed with blank output.

Trying to grab a debug log (not much time left for today but I think it'll be done in time).  I haven't tried master on L1UM v2 lately either, not sure it's from this branch.

(Also not sure the problem is here, just putting this somewhere relevant :slightly_smiling_face: )

Reply:
Cannot replicate neither on tpm2 qemu boards or nv4x :/

Reply:
My hypothesis is that you haven't tested your proposed changes merged under this pr under https://github.com/linuxboot/heads/pull/1875/commits/a06ead69bcf7e83b8f04c439ba7528a68fd1dd19 (maybe old code artifact under initrd when you tested?)

All is good on my side here and your comment block  doesn't reflect https://github.com/linuxboot/heads/pull/1875/commits/a06ead69bcf7e83b8f04c439ba7528a68fd1dd19